lib/data_update_scripts/20200910140109_cleanup_published_articles_with_duplicate_user_id_title_body_markdown.rb

```diff
@@ -0,0 +1,104 @@
+module DataUpdateScripts
+  class CleanupPublishedArticlesWithDuplicateUserIdTitleBodyMarkdown
+    def run
+      # load all published duplicates
+      rows = ActiveRecord::Base.connection.execute(
+        <<~SQL.squish,
+          SELECT *
+          FROM
+              (SELECT *,
+                      ROW_NUMBER() OVER(PARTITION BY user_id, title, body_markdown) AS row_number,
+                      COUNT(*) OVER(PARTITION BY user_id, title, body_markdown) AS num_rows_in_partition
+               FROM articles) duplicates
+          WHERE published = 't'
+              AND num_rows_in_partition > 1 -- we select all rows in each partition with duplicates
+        SQL
+      )
+
+      articles_to_delete_ids = []
+      ActiveRecord::Base.transaction do
```

Comment:
The default transaction isolation level is read committed. I assume that's what we want here, but double checking doesn't hurt.

Reply:
Would you suggest otherwise? I'm not sure we need a different one

Reply:
No, I think it's fine. I just wanted to double check, all good 👍